pandas/core/arrays/string_arrow.py

```diff
@@ -341,7 +342,19 @@ def _concat_same_type(cls, to_concat) -> ArrowStringArray:
             )
         )
 
-    def __getitem__(self, item: PositionalIndexer) -> Any:
+    @overload
+    def __getitem__(self, item: int | np.integer) -> str:
```

Comment:
i dont know this class that well; could this give pd.NA?

Reply:
Yes, fixed in next commit.